File: python/taflow/donchian_channels.py
"""Canonical descriptive alias for rolling Donchian channels."""

from typing import Any

from .donchian import Donchian


class DonchianChannels(Donchian):
    """Canonical descriptive name for the persistent Donchian channel state."""

    def append(self, high: float, low: float) -> "DonchianChannels":
        super().append(high, low)
        return self

    def extend(self, high: Any, low: Any) -> "DonchianChannels":
        super().extend(high, low)
        return self

    def reset(self) -> "DonchianChannels":
        super().reset()
        return self

__all__ = ["DonchianChannels"]

File: crates/taflow-core/src/stream/donchian_channels.rs
use super::donchian::{Donchian, DonchianValue};
use crate::error::TaResult;

#[derive(Debug, Clone)]
/// Canonical descriptive wrapper for the persistent Donchian channel state.
pub struct DonchianChannels {
    inner: Donchian,
}

impl DonchianChannels {
    pub fn new(period: usize) -> TaResult<Self> {
        Ok(Self {
            inner: Donchian::new(period)?,
        })
    }

    pub fn append(&mut self, high: f64, low: f64) -> Option<DonchianValue> {
        self.inner.append(high, low)
    }

    pub fn value(&self) -> Option<DonchianValue> {
        self.inner.value()
    }

    pub fn reset(&mut self) {
        self.inner.reset();
    }
}
